modules/azure/compute.go

```diff
@@ -8,97 +8,444 @@ import (
+	require.NoError(t, err)
+
+	return mngDiskCount
+}
+
+// GetVirtualMachineManagedDiskCountE gets the Managed Disk count of the specified Azure Virtual Machine.
+func GetVirtualMachineManagedDiskCountE(vmName string, resGroupName string, subscriptionID string) (int, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return 0, err
+	}
+
+	return len(*vm.StorageProfile.DataDisks), nil
+}
+
+// GetVirtualMachineOsDiskName gets the OS Disk name of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineOsDiskName(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
+	osDiskName, err := GetVirtualMachineOsDiskNameE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return osDiskName
+}
+
+// GetVirtualMachineOsDiskNameE gets the OS Disk name of the specified Azure Virtual Machine.
+func GetVirtualMachineOsDiskNameE(vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return "", err
 	}
 
-	// Create a VM client
-	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	return *vm.StorageProfile.OsDisk.Name, nil
+}
+
+// GetVirtualMachineState gets the State of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineState(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
+	vmState, err := GetVirtualMachineStateE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vmState
+}
+
+// GetVirtualMachineStateE gets the State of the specified Azure Virtual Machine.
+func GetVirtualMachineStateE(vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	return vm.Status, nil
+}
+
+// GetVirtualMachineAvailabilitySetID gets the Availability Set ID of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineAvailabilitySetID(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
+	avsID, err := GetVirtualMachineAvailabilitySetIDE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return avsID
+}
+
+// GetVirtualMachineAvailabilitySetIDE gets the Availability Set ID of the specified Azure Virtual Machine.
+func GetVirtualMachineAvailabilitySetIDE(vmName string, resGroupName string, subscriptionID string) (string, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	if vm.AvailabilitySet == nil {
+		return "", NewNotFoundError("Availability Set", "", vmName)
```

Comment:
If the VM is not in an availability set, that is not necessarily an error but maybe you are thinking that if someone called this function and no availability set was found, this should be an error.  Am I understanding that correctly?

Reply:
Correct again, removed error.